PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/AzureCoreToolsMissingNupkgInstaller.kt

```diff
@@ -92,71 +85,69 @@ class AzureCoreToolsMissingNupkgInstaller : StartupActivity {
-                            application.runReadAction {
-                                var keepMarker = false
-                                val fileContent = LoadTextUtil.loadText(file, 4096)
-
-                                // Check for known marker words
-                                val knownMarker = markerToTriggerMap.filter { fileContent.contains(it.key, true) }.firstOrNull()
-                                if (knownMarker != null) {
-                                    // Determine project(s) to install into
-                                    val installableProjects = WorkspaceModel.getInstance(project)
-                                            .getProjectModelEntities(file, project)
-                                            .mapNotNull { it.containingProjectEntity() }
-
-                                    if (installableProjects.isEmpty()) {
-                                        keepMarker = true
-                                    }
+                if (installableProjects.isEmpty()) return null
+
+                // For every known trigger name, verify required dependencies are installed
+                for ((triggerName, dependency) in knownMarker.value) {
+                    if (fileContent.contains(triggerName, true)) {
+                        for (installableProject in installableProjects) {
+                            val riderNuGetFacade = RiderNuGetHost.getInstance(project)
+                                    .facade
 
-                                    // For every known trigger name, verify required dependencies are installed
-                                    for ((triggerName, dependency) in knownMarker.value) {
-                                        if (fileContent.contains(triggerName, true)) {
-                                            for (installableProject in installableProjects) {
-                                                val riderNuGetFacade = RiderNuGetHost.getInstance(project)
-                                                        .facade
-
-                                                val isInstalled = riderNuGetFacade.host.nuGetProjectModel
-                                                        .projects[installableProject.getId(project)]
-                                                        ?.explicitPackages?.any { it.id.equals(dependency.id, ignoreCase = true) }
-                                                            ?: false
-
-                                                if (!isInstalled) {
-                                                    riderNuGetFacade.installForProject(
-                                                            installableProject.name, dependency.id, dependency.version)
-                                                }
-                                            }
-                                        }
+                            fun isInstalled() = riderNuGetFacade.host.nuGetProjectModel
```

Comment:
I think it worth moving out from the loop as well or move into a separate class level function